cmd/vmwrapper/vmwrapper.go

```diff
@@ -75,13 +75,42 @@ func main() {
 	// for cgroups that aren't managed by libvirt.
 	// If we don't do this, the VM pod will be killed by kubelet when Virtlet pod
 	// is removed dnd cgroup-per-qos is enabled in kubelet settings.
+	podId := os.Getenv(config.VmPodIdEnvVarName)
+	glog.V(5).Infof("Got VM POD ID: %s", podId )
+
+	computeCgroupRoot := os.Getenv(config.VmCgroupRootEnvVarName)
+	glog.V(5).Infof("Got CgroupRoot: %s", computeCgroupRoot )
+
 	cm := cgroups.NewManager(os.Getpid(), nil)
-	for _, ctl := range []string{"hugetlb", "systemd", "pids"} {
-		if _, err := cm.GetProcessController(ctl); err == nil {
-			err = cm.MoveProcess(ctl, "/")
+
+	controllers, err := cm.GetProcessControllers()
+
+	if err != nil {
+		glog.Warningf("Failed to get cgroup controllers: %v", err)
+	} else {
+		for ctl, ctlPath := range controllers {
+			glog.V(5).Infof("Cgroup controller name %s, path %s", ctl, ctlPath)
+
+			cgPath := "/"
+			// Preserve the previous behavior to put VM process to root for systemd, hugetlb and pids controllers
```

Comment:
Just be curious why we need to preserve the behavior for these three cgroup controllers.  I think the purpose of this CR is to change the behavior.   Is this because container pods did so for these three controllers?

Reply:
The change for this PR is to move the CPU, Memory etc resources to under kubepods as the root parent. the reason the vm process was kept at the root level for the PIDS controller is to ensure the VM process is running when virtlet POD/Kubelet is stopped for some reason. we will need to find a way to handle it ( as described in Arktos issue 266 )

Reply:
I don't understand - why is this different than the case of container pods?

Reply:
there is no difference from the container pods. in virtlet, the VM process was generated from the virtlet POD ( the libvirt container process ) so that the VM PID was added into the .../virtletpodCP/libvirtContainer/. in turn,  the CG Manager in kubelet scans through all PIDs under a failed POD and kill them. in this case, when virtlet DS POD restarts, the VM process will gone.

with the change in this PR, since we actually put the PIDs under the VM POD itself, the VM pod is safe with Virtlet POD failed and restart.

I have removed this check and the corresponding comments in the original comments as all.